### PR TITLE
Fix for Issue #33

### PR DESCRIPTION
**Pull request by jhbertra**
## Status
AWAITING REVIEW

## Description
Internal architecture was updated to remove deprecated features that were removed in the recent release of RC6.

`CompleterCmp` and `COMPLETER_DATA_PROVIDERS` were removed from the exports, and an NgModule was added to replace them.

Demo and readme updated with correct usage of new architecture.

## Todos
- [ ] Review

### CODE REVIEW

**Comment by jhbertra** [comment]
Tagging #33 in this PR

**Comment by oferh** [comment]
@jhbertra thanks!
I'll review and merge if all is OK

**Comment by jhbertra** [comment]
no problem.  I had to redo some internal architecture because Google literally pushed out RC 6 an hour before I started... and removed a bunch of deprecated stuff.

File: README.md [hunk 1]
```diff
@@ -14,33 +14,38 @@ Click for the [demo](http://oferh.github.io/ng2-completer/)
...
-
-bootstrap(AppComponent, [
-  HTTP_PROVIDERS,
-  disableDeprecatedForms(),
-  provideForms()
-]);
+import { NgModule } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { AppComponent } from './app-cmp';
+import { Ng2CompleterModule } from "../src/ng2-completer.module";
```
Comment (oferh) [inline README.md]: import from `ng2-completer` not from source

File: README.md [hunk 2]
```diff
@@ -14,33 +14,38 @@ Click for the [demo](http://oferh.github.io/ng2-completer/)
...
+import { FormsModule } from "@angular/forms";
```
Comment (oferh) [inline README.md]: please group all angular imports

**Comment by armno** [comment]
Hi guys, sorry for jumping in, but wouldn't it be nice if there is [syntax highlighting](https://help.github.com/articles/creating-and-highlighting-code-blocks/#syntax-highlighting) enabled for code example too to improve readability?.

for example, this
```
import { Component } from '@angular/core';
import {CompleterCmp, CompleterService, CompleterData, COMPLETER_DATA_PROVIDERS} from 'ng2-completer';

@Component({
  selector: 'my-component',
  template: `<h1>Search color</h1>
            <ng2-completer [(ngModel)]="searchStr" [dataService]="dataService" [minSearchLength]="0"></ng2-completer>`,
  directives: [CompleterCmp],
  providers: [COMPLETER_DATA_PROVIDERS, CompleterService]
})
export class MyComponent {

  private searchStr: string;
  private dataService: CompleterData;
  private searchData = [
    { color: 'red', value: '#f00' },
    { color: 'green', value: '#0f0' },
    { color: 'blue', value: '#00f' },
    { color: 'cyan', value: '#0ff' },
    { color: 'magenta', value: '#f0f' },
    { color: 'yellow', value: '#ff0' },
    { color: 'black', value: '#000' }
  ];

  constructor(private completerService: CompleterService) {
    this.dataService = completerService.local(this.searchData, 'color', 'color');
  }
}
```

would be this:
```ts
import { Component } from '@angular/core';
import {CompleterCmp, CompleterService, CompleterData, COMPLETER_DATA_PROVIDERS} from 'ng2-completer';

@Component({
  selector: 'my-component',
  template: `<h1>Search color</h1>
            <ng2-completer [(ngModel)]="searchStr" [dataService]="dataService" [minSearchLength]="0"></ng2-completer>`,
  directives: [CompleterCmp],
  providers: [COMPLETER_DATA_PROVIDERS, CompleterService]
})
export class MyComponent {

  private searchStr: string;
  private dataService: CompleterData;
  private searchData = [
    { color: 'red', value: '#f00' },
    { color: 'green', value: '#0f0' },
    { color: 'blue', value: '#00f' },
    { color: 'cyan', value: '#0ff' },
    { color: 'magenta', value: '#f0f' },
    { color: 'yellow', value: '#ff0' },
    { color: 'black', value: '#000' }
  ];

  constructor(private completerService: CompleterService) {
    this.dataService = completerService.local(this.searchData, 'color', 'color');
  }
}
```

I can create another PR just for this but as I see this PR already updates the `readme.md`, so I wanted to ask first.

**Comment by oferh** [comment]
@armno please create it as a new issue.

**Comment by oferh** [comment]
@jhbertra please resolve conflicts in the branch

**Comment by jhbertra** [comment]
Done!

**Comment by oferh** [comment]
@jhbertra Thanks!

**Pull request merged**